packages/itwinui-react/src/core/Checkbox/Checkbox.tsx

```diff
@@ -92,18 +109,24 @@ export const Checkbox = React.forwardRef((props, ref) => {
             'iui-checkbox-visibility': variant === 'eyeball',
             'iui-loading': isLoading,
           },
-          className && { [className]: !label },
+          boxClassName && { [boxClassName]: !label },
```

Comment:
I think you got confused a bit: this is the main Checkbox component - the little square, therefore, it should get "main" className, style, etc. props. 
Undo this change and check if you are able to customise `label`.

Reply:
Isn't label already being customized by WrapperProps?   wrapperProps?: React.ComponentProps<'label'>;
Or do you mean the label as in the text?